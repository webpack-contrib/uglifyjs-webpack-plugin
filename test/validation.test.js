import UglifyJsPlugin from '../src';

it('validation', () => {
  /* eslint-disable no-new */
  expect(() => {
    new UglifyJsPlugin({ test: /foo/ });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ test: [/foo/] });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ include: /foo/ });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ include: [/foo/] });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ exclude: /foo/ });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ exclude: [/foo/] });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ doesntExist: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ cache: true });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ cache: false });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ cache: 'path/to/cache/directory' });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ cache: {} });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ cacheKeys() {} });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ parallel: true });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ parallel: false });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ parallel: 2 });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ parallel: '2' });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ parallel: {} });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ sourceMap: true });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ sourceMap: false });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ sourceMap: 'true' });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ minify() {} });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: null });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: {} });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({
      uglifyOptions: {
        ecma: 5,
        warnings: false,
        parse: {},
        compress: true,
        mangle: { inline: false },
        output: { comments: /^\**!|@preserve|@license|@cc_on/ },
        toplevel: false,
        nameCache: {},
        ie8: false,
        keep_classnames: false,
        keep_fnames: false,
        safari10: false,
      },
    });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { ie8: false } });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { ie8: true } });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { ie8: 'false' } });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { emca: 5 } });
  }).not.toThrow();

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { emca: 8 } });
  }).not.toThrow();

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { ecma: 7.5 } });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { ecma: true } });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { ecma: '5' } });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { ecma: 3 } });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ uglifyOptions: { ecma: 10 } });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new UglifyJsPlugin({ extractComments: true });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ extractComments: false });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ extractComments: /comment/ });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ extractComments() {} });
  }).not.toThrow('Validation Error');

  expect(() => {
    new UglifyJsPlugin({ warningsFilter() {} });
  }).not.toThrow('Validation Error');
  /* eslint-enable no-new */
});
