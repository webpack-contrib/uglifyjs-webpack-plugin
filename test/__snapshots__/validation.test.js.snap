// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"UglifyJs Plugin Invalid Options

options['doesntExist'] is an invalid additional property
"
`;

exports[`validation 2`] = `
"UglifyJs Plugin Invalid Options

options.cache should be boolean
options.cache should be string
options.cache should match exactly one schema in oneOf
"
`;

exports[`validation 3`] = `
"UglifyJs Plugin Invalid Options

options.parallel should be boolean
options.parallel should be integer
options.parallel should match exactly one schema in oneOf
"
`;

exports[`validation 4`] = `
"UglifyJs Plugin Invalid Options

options.parallel should be boolean
options.parallel should be integer
options.parallel should match exactly one schema in oneOf
"
`;

exports[`validation 5`] = `
"UglifyJs Plugin Invalid Options

options.sourceMap should be boolean
"
`;

exports[`validation 6`] = `
"UglifyJs Plugin Invalid Options

options.uglifyOptions should be object
"
`;

exports[`validation 7`] = `
"UglifyJs Plugin Invalid Options

options.uglifyOptions.ie8 should be boolean
"
`;

exports[`validation 8`] = `
"UglifyJs Plugin Invalid Options

options.uglifyOptions.ecma should be integer
"
`;

exports[`validation 9`] = `
"UglifyJs Plugin Invalid Options

options.uglifyOptions.ecma should be integer
"
`;

exports[`validation 10`] = `
"UglifyJs Plugin Invalid Options

options.uglifyOptions.ecma should be integer
"
`;

exports[`validation 11`] = `
"UglifyJs Plugin Invalid Options

options.uglifyOptions.ecma should be >= 5
"
`;

exports[`validation 12`] = `
"UglifyJs Plugin Invalid Options

options.uglifyOptions.ecma should be <= 8
"
`;
